perf/lua/1mops_write.lua

```diff
@@ -0,0 +1,294 @@
+-- a boolean option. The options list with default values:
+--   engine <string, 'memtx'> - select engine type
+--   fibers <number, 50> - number of fibers to run simultaneously
+--   index <string, 'HASH'> - select index type
+--   nodes <number, 1> - number of nodes to test one-way replication
+--   nohint <boolean> - to turn the hints off
+--   ops <number, 10000000> - total amount of replaces to be performed
+--   sync <boolean> - to turn the synchro replication on
+--   transaction <number, 150> - number of replaces in one transaction
+--   warmup <number, 10> - percent of ops to skip before measurement
+--
+-- Being run without options tests the HASH index of memtx engine by
+-- running of 50 fibers that replace 150 integers using each as a primary
+-- key. Refer to the 'fiber_load' function below to change the workload.
+
+local err, popen = pcall(require,'popen')
+if err ~= nil then
+    popen = {}
+end
+local clock = require('clock')
+local fiber = require('fiber')
+local math = require('math')
+local json = require('json')
+
+-- working dir is cleared/created for every run, set test_dir
+-- during the test a subdir will be created for each replica
+local function getarg(s, f)
+    for i = 1, #arg do
+        if arg[i] == '--'..s or arg[i] == '-'..s then
+            if f == nil then return true
+            else return f(arg[i+1]) or error('wrong option -' .. s)
+            end
+        end
+    end
+    return false
+end
+
+local test_dir = './relay-test.data'
+
+-- turn true to test the qsync
+local test_qsync = getarg('sync')
+
+-- number of operations performed by test
+local num_ops = getarg('ops', tonumber) or 10000000
+
+-- number of fibers
+local num_fibers = getarg('fibers', tonumber) or 50
+
+-- number of operations per transaction
+local ops_per_txn = getarg('transaction', tonumber) or 150
+
+-- number of nodes - master and replicas
+local nodes = getarg('nodes', tonumber) or 1
+
+-- type of index to use
+local index_config = {}
+index_config.type = getarg('index', tostring) or 'HASH'
+if getarg('nohint') then index_config.hint = false end
+
+-- engine to use
+local engine = getarg('engine', tostring) or 'memtx'
+assert(engine == 'memtx' or engine == 'vinyl')
+
+local warmup_thr = getarg('warmup', tonumber) or 10
+warmup_thr = warmup_thr > 100 and 100 or warmup_thr
+warmup_thr = warmup_thr < 0 and 0 or warmup_thr
+-- END OF TUNABLE OPTIONS
+
+-- transactions per fiber
+local trans_per_fiber = num_ops/ops_per_txn/num_fibers
+
+-- by default no output from replicas are received
+-- redirect it into master's one breaks terminal
+local std_redirect = {
+    stdin='devnull',
+    stdout='devnull',
+    stderr='devnull'
+}
+
+print(string.format([[
+making %d operations,
+%d operations per txn usnig %d fibers,
+replicaset of %d nodes]],
+      num_ops, ops_per_txn, num_fibers, nodes))
+
+-- cleanup the test dir - handy for restart
+os.execute('rm -rf ' .. test_dir)
+os.execute('mkdir ' .. test_dir)
```

Comment:
I believe you can simply use [os.tmpname](https://www.lua.org/manual/5.1/manual.html#pdf-os.tmpname) here, instead of such kind of cleanup?

Reply:
Then it will pile up the temporary files from sequential runs? At least for those crashed, where I can't handle all type of exits or make any on_shutdown() operable. Removing the work_dir right after box_cfg() will work only until xlog rotation, where new file should be open in this folder.

Reply:
What are the critical points where the benchmark can crash?

Reply:
Well, beyond other issues, the very beginning of the test
```{lua}
[s.ostanevich@dev1 lua]$ ../../src/tarantool -i
tarantool> box.cfg{work_dir='asdf'}
can't chdir to `asdf': No such file or directory
[s.ostanevich@dev1 lua]$ echo $?
1
[...]
2023-07-26 13:27:47.084 [52922] main/103/interactive F> failed to preallocate 4398046511104 bytes: Cannot allocate memory, check option 'memtx_memory' in box.cfg(..)
[s.ostanevich@dev1 lua]$ echo $?
1
```
Literally - any problem in box.cfg{} can lead to an immediate termination.

Reply:
Unfortunately, the `os.tmpname()` creates a temporary file, so as `io.tmpfile()`. Apparently, there's no safe interface to provide a unique name that will be available and unique over time between its creation and using it.
The only w/a I can envision is to create the temp file, then extend its name with a suffix and create a temp dir along with it, cleaning up both upon completion. Upvote if it is feasible.

Reply:
Nevermind, will follow @ligurio proposal of `fio.tmpdir()`